modules/generators/content/password_file/secgen_local/local.rb

```diff
@@ -5,22 +5,97 @@
 class PasswordFileGenerator < StringGenerator
   attr_accessor :website_sample
   attr_accessor :username_sample
-  attr_accessor :pass_sample
+  attr_accessor :pass_list
+  attr_accessor :password_1
+  attr_accessor :password_2
+  attr_accessor :password_3
+  attr_accessor :password_4
+  attr_accessor :password_5
   LOCAL_DIR = File.expand_path('../../',__FILE__)
   TEMPLATE_PATH = "#{LOCAL_DIR}/templates/password_file.md.erb"
 
-  def generate
+  def initialize
+    super
+    self.password_1 = ''
+    self.password_2 = ''
+    self.password_3 = ''
+    self.password_4 = ''
+    self.password_5 = ''
+  end
+  
+  def get_options_array
+    super + [['--password1', GetoptLong::OPTIONAL_ARGUMENT],
+             ['--password2', GetoptLong::OPTIONAL_ARGUMENT],
+             ['--password3', GetoptLong::OPTIONAL_ARGUMENT],
+             ['--password4', GetoptLong::OPTIONAL_ARGUMENT],
+             ['--password5', GetoptLong::OPTIONAL_ARGUMENT]]
+  end
+
+  def process_options(opt, arg)
+    super
+    case opt
+    when '--password1'
+      self.password_1 << arg;
+    when '--password2'
+      self.password_2 << arg;
+    when '--password3'
+      self.password_3 << arg;
+    when '--password4'
+      self.password_4 << arg;
+    when '--password5'
+      self.password_5 << arg;
+    end
+  end
+
+def generate
+  pass_list = Array.new
+  passCount = 5
+
+  if self.password_1 != ''
```

Comment:
Use pass_list as the input array, and randomly reuse passwords if we don't have enough provided. For example, you could have just one leaked_password as input, and just feed it into every password in the list
```
# repeat a password, if we are not given enough passwords
if pass_list.size < 5
  pass_list << pass_list.sample
end
```